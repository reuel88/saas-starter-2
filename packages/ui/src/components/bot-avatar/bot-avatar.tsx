import { Avatar, AvatarImage } from "../ui/avatar";
import { FC } from "react";
import { cn } from "@repo/utilities";

type BotAvatarProps = {
  className?: string;
  imageSrc?: string;
};

export const BotAvatar: FC<BotAvatarProps> = ({
  className = "",
  imageSrc = "logo.png",
}) => {
  return (
    <Avatar className={cn("h-8 w-8", className)}>
      <AvatarImage src={imageSrc} />
    </Avatar>
  );
};
