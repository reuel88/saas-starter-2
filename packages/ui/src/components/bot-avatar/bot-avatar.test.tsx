import { render, screen } from "@testing-library/react";
import { BotAvatar } from "./bot-avatar";

describe("UserAvatar", () => {
  const originalGlobalImage = window.Image;

  beforeAll(() => {
    (window.Image as any) = class MockImage {
      onload: () => void = () => {};
      src: string = "";
      constructor() {
        setTimeout(() => {
          this.onload();
        }, 300);
        return this;
      }
    };
  });

  afterAll(() => {
    window.Image = originalGlobalImage;
  });

  it("should render an Avatar component with a default image source when no imageSrc prop is provided", async () => {
    // Arrange
    const expectedSrc = "logo.png";

    // Act
    const { baseElement } = render(<BotAvatar />);
    const avatarImage = await screen.findByRole("img");

    // Assert
    expect(avatarImage).toHaveAttribute("src", expectedSrc);
    expect(baseElement).toMatchSnapshot();
  });

  it("should render an BotImage component with the provided image source", async () => {
    // Arrange
    const imageSrc = "https://github.com/shadcn.png";

    // Act
    const { baseElement } = render(<BotAvatar imageSrc={imageSrc} />);
    const avatarImage = await screen.findByRole("img");

    // Assert
    expect(avatarImage).toHaveAttribute("src", imageSrc);
    expect(baseElement).toMatchSnapshot();
  });
});
