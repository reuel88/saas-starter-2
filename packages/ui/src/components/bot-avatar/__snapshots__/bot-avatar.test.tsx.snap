// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`UserAvatar > should render an Avatar component with a default image source when no imageSrc prop is provided 1`] = `
<body>
  <div>
    <span
      class="relative flex shrink-0 overflow-hidden rounded-full h-8 w-8"
    >
      <img
        class="aspect-square h-full w-full"
        src="logo.png"
      />
    </span>
  </div>
</body>
`;

exports[`UserAvatar > should render an BotImage component with the provided image source 1`] = `
<body>
  <div>
    <span
      class="relative flex shrink-0 overflow-hidden rounded-full h-8 w-8"
    >
      <img
        class="aspect-square h-full w-full"
        src="https://github.com/shadcn.png"
      />
    </span>
  </div>
</body>
`;
